private/bufpkg/bufmodule/bufmodulestore/module_data_store.go

```diff
@@ -69,20 +72,23 @@ func ModuleDataStoreWithTar() ModuleDataStoreOption {
 /// *** PRIVATE ***
 
 type moduleDataStore struct {
-	logger *zap.Logger
-	bucket storage.ReadWriteBucket
+	logger         *zap.Logger
+	clientProvider bufapi.ClientProvider
+	bucket         storage.ReadWriteBucket
 
 	tar bool
 }
 
 func newModuleDataStore(
 	logger *zap.Logger,
+	clientProvider bufapi.ClientProvider,
```

Comment:
I would actually not pass it to the moduleDataStore, and instead add a comment that we expect all lock files in the moduleDataStore to have digests. We control this.